### PR TITLE
Baseline variable renames

### PR DESCRIPTION
**Pull request by tonowak**
This PR is stacked on https://github.com/obi1kenobi/cargo-semver-checks/pull/348.

In the next PR, the trait `BaselineLoader` and its structs will be also user for the `current` crate, the names won't be relevant anymore. This PR renames the trait and structs.

### CODE REVIEW

File: src/rustdoc_gen.rs [hunk 1]
```diff
@@ -441,13 +468,13 @@ fn bytes2str(b: &[u8]) -> &std::ffi::OsStr {
     std::ffi::OsStr::new(str::from_utf8(b).unwrap())
 }
 
-pub(crate) struct RegistryBaseline {
+pub(crate) struct RustdocFromRegistry {
     target_root: PathBuf,
     version: Option<semver::Version>,
     index: crates_index::Index,
 }
```
Comment (tonowak) [inline src/rustdoc_gen.rs:475]: Is there a way to add a `derive Debug` to this struct when `index` doesn't implement this trait? Like, implementing the `Debug` for all variables besides `index`?

Comment (obi1kenobi) [reply src/rustdoc_gen.rs:475]: Yes, you just need to write a manual `impl Debug for RustdocFromRegistry`.

**Comment by obi1kenobi** [comment]
This is good, just resolve the merge conflict and add the manual `Debug` impl and it's good to go.

File: src/rustdoc_gen.rs (hunk 1 above)
Comment (tonowak) [reply src/rustdoc_gen.rs:472]: Done in 4113d4f07681e03a09c98c8edd9e65b0d52c98d8.

File: src/rustdoc_gen.rs [hunk 2]
```diff
@@ -464,13 +465,22 @@ fn bytes2str(b: &[u8]) -> &std::ffi::OsStr {
...
 
-impl RegistryBaseline {
+impl core::fmt::Debug for RustdocFromRegistry {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("RustdocFromRegistry")
+            .field("target_root", &self.target_root)
+            .field("version", &self.version)
+            .finish()
```
Comment (obi1kenobi) [inline src/rustdoc_gen.rs:479]:
```suggestion
            .field("version", &self.version)
            .field("index", "<elided>")
            .finish()
```

**Review by obi1kenobi — APPROVED** [review]

**Comment by tonowak** [comment]
I'll fix the compile error.

**Comment by obi1kenobi** [comment]
Thanks! I wish GitHub ran rust-analyzer in WASM :)

**Pull request merged**